sdk/lib/opentelemetry/sdk/resources/detectors/google_cloud_platform.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'google-cloud-env'
+
+module OpenTelemetry
+  module SDK
+    module Resources
+      module Detectors
+        module GoogleCloudPlatform
+          extend self
+
+          def detect
+            gcp_env = Google::Cloud::Env.new
+            return Resource.create unless gcp_env.compute_engine?
+
+            resource_labels = {}
+            resource_labels[CLOUD_RESOURCE[:provider]] = 'gcp'
+            resource_labels[CLOUD_RESOURCE[:account_id]] = gcp_env.project_id || ''
```

Comment:
This deserves a second set of eyes I'm not totally sure this is the correct value being set here, I was following the work done on the JS library https://github.com/open-telemetry/opentelemetry-js/pull/899/files#diff-2ae8d53e5ba7424080459738d6bc210bR45

From the spec 
>The cloud account id used to identify different entities.

But this is using `project-id` from https://cloud.google.com/compute/docs/storing-retrieving-metadata#default

Reply:
`project-id` seems wrong, but the metadata doesn't supply the account ID. I would suggest leaving this as-is and opening an issue in the spec repo asking for clarification.